ocaml/xapi/xapi_pool.ml

```diff
@@ -1563,6 +1563,42 @@ let join_common ~__context ~master_address ~master_username ~master_password
             "Unable to set the write the new pool certificates to the disk : %s"
             (ExnHelper.string_of_exn e)
       ) ;
+      (* Sync RPM GPG keys to every host of the current pool *)
+      ((* Retrieve keys from the new pool *)
+       let pubkeys =
```

Comment:
This won't be enough to ensure all hosts have all the keys, we used to do this for pool certificates and it didn't work for all cases.

Consider the case where multiple host are joining a pool at the same time. The first certificate to join will miss the gpg keys of successive joiners before it's able to restart and connect to the pool coordinator as part of the pool.

On pull start supporters needs to import the missing keys from the pool coordinator to be able to solve this race condition

See https://github.com/xapi-project/xen-api/commit/c6ae148d024a6158d3922e0c6fafcddcdc137073 to see how it was solved for pool certificates

Reply:
Unlike certificates in which case there are certs exchanging between hosts, the GPG keys (at least RPM GPG public keys) would be synced from coordinator to other members in uni-direction.
But indeed, there may be cases that a key was not synced to a member host. So I appended a commit for the member host to pull the key from coordinator on-demand.

Reply:
> Unlike certificates in which case there are certs exchanging between hosts, the GPG keys (at least RPM GPG public keys) would be synced from coordinator to other members in uni-direction.

On pool join the joiner downloads the certificates from the coordinator, and it pushes its certificates to the coordinator, which will push it to the pool supporters. The race condition is the same as here

Reply:
Discussed offline. The joiner should remove the managed GPG keys in old pool before joining.

Reply:
Appended [113c8ea](https://github.com/xapi-project/xen-api/pull/4825/commits/113c8ea648a6d895d7be373cf4d17316e33a0571) for this.